src/pk/ecc/ecc_verify_hash.c

```diff
@@ -65,12 +84,19 @@ static int _ecc_verify_hash(const unsigned char *sig,  unsigned long siglen,
       if ((err = mp_read_unsigned_bin(r, (unsigned char *)sig,   i)) != CRYPT_OK)                       { goto error; }
       if ((err = mp_read_unsigned_bin(s, (unsigned char *)sig+i, i)) != CRYPT_OK)                       { goto error; }
    }
+   else if (sigformat == LTC_ECCSIG_ETH27) {
+      /* Ethereum (v,r,s) format */
+      if (siglen != 65) { /* Only secp256k1 curves */
```

Comment:
There's no indication of the curve _at all_, as with the RFC7518, but since this signature is used only by applications using the secp256k1 curve, it must be 65 bytes long. If it's not, it's some other format.

Reply:
When talking about RFC7518 I had a similar discussion about what we now call `LTC_ECCSIG_RFC7518` - see https://github.com/DCIT/perl-CryptX/issues/45

The thing is that by `LTC_ECCSIG_RFC7518` we mean **only the signature format** not other conditions defined by the RFC in question.

Similarly by `LTC_ECCSIG_ETH27` we should mean **the format** how the final byte sequence is constructed.

So for `LTC_ECCSIG_ETH27` we should IMO only test that `siglen` is equal to `key->dp.size * 2 + 1` and that `sig[key->dp.size * 2] >= 27`.

Reply:
The signature format in RFC7518 is at least _defined_ for different key sizes, whereas the Ethereum stuff isn't as there is only a single curve in use. There are literally no signatures in this format which aren't either 65 bytes (or, equivalently, 130 hex digits).

It's a simple enough change if you want me to make it, though it probably should also get rid of the curve OID check in `ecc_sign_hash` so there's at least some way to generate the non-standard forms. Just let me know...

Reply:
Does `ecc_recover_key` work also with other curves - especially those with `cofactor != 1`?

Reply:
Yes. The main one to test with is `secp112r2` (cofactor 4) - I've seen recovery values for that during the tests coming through as (e.g.) 6. That's with an X9.62 signature and a separate recovery value, though.

... which is a good point. Adding 27 would get rather weird for values of `v` other than 0/1.